client/state/imports/site-importer/actions.js

```diff
@@ -182,32 +183,60 @@ export const validateSiteIsImportable =
 		);
 		dispatch( startSiteImporterIsSiteImportable() );
 
-		return wpcom.req
-			.get( {
-				path: `/sites/${ siteId }/site-importer/is-site-importable?${ stringify( params ) }`,
-				apiNamespace: 'wpcom/v2',
-			} )
-			.then( ( response ) => {
-				dispatch(
-					recordTracksEvent( 'calypso_site_importer_validate_site_success', {
-						blog_id: siteId,
-						site_url: response.site_url,
-						supported_content: sortAndStringify( response.supported_content ),
-						unsupported_content: sortAndStringify( response.unsupported_content ),
-						site_engine: response.engine,
-					} )
-				);
-				dispatch( siteImporterIsSiteImportableSuccessful( response ) );
-			} )
-			.catch( ( error ) => {
+		const isSiteImportable = wpcom.req.get( {
+			path: `/sites/${ siteId }/site-importer/is-site-importable?${ stringify( params ) }`,
+			apiNamespace: 'wpcom/v2',
+		} );
+
+		const analyzeUrl = wpcom.req.get( {
+			path: `/imports/analyze-url?${ stringify( { site_url: targetSiteUrl } ) }`,
+			apiNamespace: 'wpcom/v2',
+		} );
+
+		const [ isSiteImportableResult, analyzeUrlResult ] = await Promise.allSettled( [
+			isSiteImportable,
+			analyzeUrl,
+		] );
+
+		if ( isSiteImportableResult.status === 'fulfilled' ) {
+			const response = isSiteImportableResult.value;
+			dispatch(
+				recordTracksEvent( 'calypso_site_importer_validate_site_success', {
+					blog_id: siteId,
+					site_url: response.site_url,
+					supported_content: sortAndStringify( response.supported_content ),
+					unsupported_content: sortAndStringify( response.unsupported_content ),
+					site_engine: response.engine,
+				} )
+			);
+			dispatch( siteImporterIsSiteImportableSuccessful( response ) );
+		} else {
+			dispatch(
+				recordTracksEvent( 'calypso_site_importer_validate_site_fail', {
+					blog_id: siteId,
+					site_url: targetSiteUrl,
+				} )
+			);
+
+			// do platform validation if param - targetPlatform is given
+			if (
+				isSiteImportableResult?.reason?.code === 1000002 &&
```

Comment:
Shouldn't we modify the errors in the backend instead of reading the code and modifying the error string in the frontend? See `wp-content/lib/imports/site-importer/site-importer.php$457-467`. The backend generates a somewhat scary error message with a UUID inside. It's the same string for various types of errors. What you think @vishnugopal?

Reply:
Yeah, I considered it before. I didn't choose this approach because this error is related to the user's intent (selecting the Wix importer to import content via a WordPress.com URL). I was thinking adding a param related to a user's intent to the is-site-importable API is a bit weird. 🤔

Reply:
Hmm I can see the benefits of both approaches, but since we have decided to go with the frontend anyways (& not modify `is-site-importable` to call `analyze-url` in the backend) I'm ok with this approach for now!

Reply:
So, it's ok for me too.